### PR TITLE
Don't include text bubbles in touching queries

### LINKED ISSUE #577: Text bubbles are counted for "touching color" purposes in 3.0, but not in 2.0
In Scratch 2.0, text bubbles don't count for the purposes of "touching color" blocks, e.g. if a sprite is touching a text bubble, that doesn't count as touching white or grey or any other color on that text bubble; they are ignored completely.

In Scratch 3.0, however, "touching color" blocks can return true if the sprite is touching a text bubble's color.

This raises the question: should this be noted as a compatibility change between 2.0 and 3.0 and the current behavior kept, or should the behavior be changed to match 2.0?

<details>
<summary>
A unit test was supposed to verify that this is the case in 3.0 as well, but...
</summary>

Due to a series of weird coincidences and oversights, this unit test has erroneously passed when it should not have, and text bubbles have been part of "touching color" tests for, as far as I can tell, the entire 3.0 lifespan.
- The unit test runs on the very first frame. Unfortunately, the SVG-based text bubbles implemented when the test was first created had a delay of about 1-2 frames before showing up.
- The canvas-based text bubbles introduced in #451 are created synchronously, and so they should have revealed the error, however...
- Due to the same issue as in #544, text bubble silhouettes are not created until the bubbles have been rendered. This wouldn't be a problem, since this bug only manifests on the CPU path, but...
- Since the CPU/GPU parity testing in #419 was reverted before #544 was merged, the test never runs on the GPU path.

</details>

### PR DESCRIPTION
**Pull request by adroitwhiz**
### Resolves

Resolves https://github.com/LLK/scratch-render/issues/577

### Proposed Changes

This changes `_candidatesTouching` to exclude text bubbles, and fixes text bubbles to properly update their silhouettes.

### Reason for Changes

* Change `_candidatesTouching` to exclude text bubbles
  * This changes the behavior of all "touching" blocks to not count text bubbles.
* Add `updateSilhouette` to `TextBubbleSkin`
  * This is why the test *wasn't* failing when it should have. It was checking if it was touching the text bubble on the very first frame, before the text bubble was even rendered. At that point, the text bubble's silhouette didn't exist, as it was only generated when the text bubble was rendered. Now, the text bubble's silhouette is generated whenever `updateSilhouette` is called.

### Test Coverage

Re-enabled the "text bubbles are not touched" test

### CODE REVIEW

File: src/TextBubbleSkin.js [hunk 1]
```diff
@@ -241,6 +241,11 @@ class TextBubbleSkin extends Skin {
         this._renderedScale = scale;
     }
 
+    updateSilhouette (scale = [100, 100]) {
```
Comment (fsih) [inline src/TextBubbleSkin.js:244]: Why was this needed?

Comment (adroitwhiz) [reply src/TextBubbleSkin.js:244]: This ensures that text bubbles' "silhouettes" behave consistently on the CPU render path. Since text bubbles are no longer rendered on the CPU path, this is technically not necessary but should prevent future weirdness like what happened with the test here where text bubbles appeared to not count as being touched for the first frame.

**Review by fsih — APPROVED** [review]
LG, thanks!
